### PR TITLE
Fixed sending duplicate emails 

### PR DESCRIPTION
**Pull request by tarunnjoshi**
Fixed sending duplicate emails

### CODE REVIEW

File: wp-content/civi-extensions/goonjcustom/api/v3/Goonjcustom/CollectionCampCron.php [hunk 1]
```diff
@@ -65,6 +65,9 @@ function civicrm_api3_goonjcustom_collection_camp_cron($params) {
     $endDate = new DateTime($camp['Collection_Camp_Intent_Details.End_Date']);
     $collectionCampId = $camp['id'];
     $endDateFormatted = $endDate->format('Y-m-d');
+    $emailSentFlag = $camp['Logistics_Coordination.Email_Sent'];
```
Comment (pokhiii) [inline wp-content/civi-extensions/goonjcustom/api/v3/Goonjcustom/CollectionCampCron.php:68]:
```suggestion
    $logisticEmailSent = $camp['Logistics_Coordination.Email_Sent'];
```

File: wp-content/civi-extensions/goonjcustom/api/v3/Goonjcustom/CollectionCampCron.php [hunk 2]
```diff
@@ -65,6 +65,9 @@ function civicrm_api3_goonjcustom_collection_camp_cron($params) {
...
+    $feedbackEmailSentFlag = $camp['Logistics_Coordination.Feedback_Email_Sent'];
```
Comment (pokhiii) [inline wp-content/civi-extensions/goonjcustom/api/v3/Goonjcustom/CollectionCampCron.php:69]:
```suggestion
    $feedbackEmailSent = $camp['Logistics_Coordination.Feedback_Email_Sent'];
```

File: wp-content/civi-extensions/goonjcustom/api/v3/Goonjcustom/CollectionCampCron.php [hunk 3]
```diff
@@ -97,37 +100,63 @@ function civicrm_api3_goonjcustom_collection_camp_cron($params) {
...
 
-    // Only send the email if the end date is exactly today.
-    if ($endDateFormatted <= $todayFormatted) {
-      $mailParams = [
-        'subject' => 'Volunteer Feedback Form',
-        'from' => $fromEmail['label'],
-        'toEmail' => $contactEmailId,
-        'replyTo' => $fromEmail['label'],
-        'html' => goonjcustom_collection_camp_volunteer_feedback_email_html($organizingContactName, $collectionCampId),
+    if ($feedbackEmailSentFlag !== 1) {
```
Comment (pokhiii) [inline wp-content/civi-extensions/goonjcustom/api/v3/Goonjcustom/CollectionCampCron.php:108]:
```suggestion
    if (!$feedbackEmailSentFlag) {
```

File: wp-content/civi-extensions/goonjcustom/api/v3/Goonjcustom/CollectionCampCron.php [hunk 4]
```diff
@@ -97,37 +100,63 @@ function civicrm_api3_goonjcustom_collection_camp_cron($params) {
...
 
-    // Only send the email if the end date is lower than today.
-    if ($endDateFormatted <= $todayFormatted) {
-      $mailParams = [
-        'subject' => 'Collections Completion Notification',
-        'from' => 'urban.ops@goonj.org',
-        'toEmail' => $emailId,
-        'replyTo' => 'urban.ops@goonj.org',
-        'html' => goonjcustom_collection_camp_email_html($contactName, $collectionCampId, $recipientId, $collectionCampGoonjOffice),
+    if ($emailSentFlag !== 1) {
```
Comment (pokhiii) [inline wp-content/civi-extensions/goonjcustom/api/v3/Goonjcustom/CollectionCampCron.php:133]:
```suggestion
    if (!$emailSentFlag) {
```

File: wp-content/civi-extensions/goonjcustom/api/v3/Goonjcustom/CollectionCampCron.php [hunk 5]
```diff
@@ -97,37 +100,63 @@ function civicrm_api3_goonjcustom_collection_camp_cron($params) {
...
-      ];
-      $result = CRM_Utils_Mail::send($mailParams);
+        ];
+        $campNotificationResult = CRM_Utils_Mail::send($mailParams);
+
+        if ($campNotificationResult) {
+          // Collect the camp ID.
+          $campIdsToUpdate[] = $collectionCampId;
+
+          $results = EckEntity::update('Collection_Camp', TRUE)
```
Comment (pokhiii) [inline wp-content/civi-extensions/goonjcustom/api/v3/Goonjcustom/CollectionCampCron.php:150]:
```suggestion
          EckEntity::update('Collection_Camp', TRUE)
```

File: wp-content/civi-extensions/goonjcustom/api/v3/Goonjcustom/CollectionCampCron.php [hunk 6]
```diff
@@ -97,37 +100,63 @@ function civicrm_api3_goonjcustom_collection_camp_cron($params) {
...
-      ];
-      $result = CRM_Utils_Mail::send($mailParams);
+        ];
+        $feedbackNotificationResult = CRM_Utils_Mail::send($mailParams);
+
+        if ($feedbackNotificationResult) {
+          // Collect the initiator ID.
+          $initiatorIdsToUpdate[] = $initiatorId;
+
+          $results = EckEntity::update('Collection_Camp', TRUE)
```
Comment (pokhiii) [inline wp-content/civi-extensions/goonjcustom/api/v3/Goonjcustom/CollectionCampCron.php:125]:
```suggestion
          EckEntity::update('Collection_Camp', TRUE)
```

File: wp-content/civi-extensions/goonjcustom/api/v3/Goonjcustom/CollectionCampCron.php [hunk 7]
```diff
@@ -97,37 +100,63 @@ function civicrm_api3_goonjcustom_collection_camp_cron($params) {
...
-        'from' => 'urban.ops@goonj.org',
-        'toEmail' => $emailId,
-        'replyTo' => 'urban.ops@goonj.org',
-        'html' => goonjcustom_collection_camp_email_html($contactName, $collectionCampId, $recipientId, $collectionCampGoonjOffice),
+    if ($emailSentFlag !== 1) {
+      // Only send the email if the end date is lower than today.
+      if ($endDateFormatted <= $todayFormatted) {
+        $mailParams = [
+          'subject' => 'Collections Completion Notification',
+          'from' => 'urban.ops@goonj.org',
```
Comment (pokhiii) [inline wp-content/civi-extensions/goonjcustom/api/v3/Goonjcustom/CollectionCampCron.php:138]: can we update here too?

File: wp-content/civi-extensions/goonjcustom/api/v3/Goonjcustom/CollectionCampCron.php [hunk 8]
```diff
@@ -97,37 +100,63 @@ function civicrm_api3_goonjcustom_collection_camp_cron($params) {
...
+          'toEmail' => $emailId,
+          'replyTo' => 'urban.ops@goonj.org',
```
Comment (pokhiii) [inline wp-content/civi-extensions/goonjcustom/api/v3/Goonjcustom/CollectionCampCron.php:140]: and here

**Review by pokhiii — APPROVED** [review]

**Comment by tarunnjoshi** [comment]
Please don't merge this PR. After testing it in production, we can proceed with merging it into the production environment.

**Comment by pokhiii** [comment]
closing this in favor of https://github.com/ColoredCow/goonj/pull/304

**Pull request closed without merging**